Sources/CADCacheAdvance/CADCacheAdvance.swift

```
@@ -0,0 +1,88 @@
+/// This cache is intended to be written to and read from using the same serial queue.
+/// - Attention: This type is meant to be used by Objective-C code, and is not exposed to Swift. Swift code should use CacheAdvance<T>.
+@objc(CADCacheAdvance)
+@available(swift, obsoleted: 1.0)
+public final class __ObjectiveCCompatibleCacheAdvanceWithGenericData: NSObject {
+
+    // MARK: Initialization
+
+    /// Creates a new instance of the receiver.
+    ///
+    /// - Parameters:
+    ///   - fileURL: The file URL indicating the desired location of the on-disk store. This file should already exist.
+    ///   - maximumBytes: The maximum size of the cache, in bytes. Logs larger than this size will fail to append to the store.
+    ///   - shouldOverwriteOldMessages: When `true`, once the on-disk store exceeds maximumBytes, new entries will replace the oldest entry.
+    ///
+    /// - Warning: `maximumBytes` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache will prevent appending new messages to this cache.
+    /// - Warning: `shouldOverwriteOldMessages` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache will prevent appending new messages to this cache.
+    @objc
+    public init(
+        fileURL: URL,
+        maximumBytes: Bytes,
+        shouldOverwriteOldMessages: Bool)
+        throws
+    {
+        cache = try CacheAdvance<Data>(
+            fileURL: fileURL,
+            maximumBytes: maximumBytes,
+            shouldOverwriteOldMessages: shouldOverwriteOldMessages)
+    }
+
+    // MARK: Public
+
+    @objc
+    public var fileURL: URL {
+        cache.fileURL
+    }
+
+    /// Checks if the all the header metadata provided at initialization matches the persisted header. If not, the cache is not writable.
+    /// - Returns: `true` if the cache is writable.
+    @objc
+    public var isWritable: Bool {
+        (try? cache.isWritable()) ?? false
```

If we can't determine if a file is writable, then it isn't writable. Handling errors in Objective-C is not great, so consolidating to a simple `Bool` answer seems reasonable.

---

I think I've suggested making this change in Swift too. I think there's a valid POV that it's either writable or it isn't, and if it isn't it doesn't matter why it isn't.
But I also like not hiding errors.
I wonder if there is a way where we can make some of the Swift methods a bit more ergonomic while still giving you an option to see the full reasons (by calling into a more verbose method that throws) if you want.
But this is for another day.